awx/api/serializers.py

```diff
@@ -1739,7 +1739,9 @@ def validate(self, attrs):
             vars_dict = parse_yaml_or_json(variables)
             vars_dict['ansible_ssh_port'] = port
             attrs['variables'] = json.dumps(vars_dict)
-
+        if Group.objects.filter(name=name, inventory=attrs.get('inventory')).exists():
```

Comment:
I think you may need to check `self.instance.inventory` for the case that someone changes the name of an existing group. Similar to how below you did `attrs.get('name', self.instance and self.instance.name or '')`, a similar thing for "inventory" as opposed to "name"